fixieai/cli/auth/commands.py

```diff
@@ -0,0 +1,31 @@
+import click
+
+from fixieai.cli.auth import oauth_flow
+from fixieai.cli.auth import user_config
+
+
+def validate_not_authed(ctx, param, force):
+    if force:
+        # --force is set
+        return
+    try:
+        config = user_config.load_config()
+        if config.fixie_api_key is not None:
+            click.echo("Already authenticated. Set --force to force re-authentication.")
+            ctx.exit()
+    except FileNotFoundError:
+        pass
+
+
+@click.command("auth", help="Authorizes `fixie` to access Fixie platform.")
+@click.option("--force", is_flag=True, help="Forces authentication, even if the user is authenticated.",
+              callback=validate_not_authed, expose_value=False)
+def auth():
+    fixie_api_token = oauth_flow.oauth_flow()
+    try:
+        config = user_config.load_config()
+    except FileNotFoundError:
+        config = user_config.UserConfig()
+
+    config.fixie_api_key = fixie_api_token
+    user_config.save_config(config)
```

Comment:
I wonder if it would not be simpler to just store the API token in the system keyring, using the Python keyring library:

https://pypi.org/project/keyring/

Reply:
that seems to be suitable for passwords and things that we want the OS to protect I think. Here, we are following the pattern of `gh` and `gcloud`. They store their keys at `~/.config/gh/hosts.yml` and `~/.config/gcloud/access_tokens.db`, respectively.